Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/Saml2AllowNoSubjectAssertionSpecValidator.java

```diff
@@ -26,45 +26,66 @@
  */
 package gov.hhs.fha.nhinc.callback.cxf;
 
-import org.opensaml.saml2.core.Assertion;
-import org.opensaml.xml.validation.ValidationException;
+
+import org.apache.wss4j.common.ext.WSSecurityException.ErrorCode;
+
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.apache.wss4j.common.saml.SamlAssertionWrapper;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Checks {@link org.opensaml.saml2.core.Assertion} for Spec compliance. This validator relaxes the rules by not
  * checking for the existence of Subject when the Assertion contains AttributeStatements. This is required for
  * interoperability with previous CONNECT gateways.
  */
 public class Saml2AllowNoSubjectAssertionSpecValidator extends Saml2ExchangeAuthFrameworkValidator {
-
+    private static final Logger LOG = LoggerFactory.getLogger(Saml2AllowNoSubjectAssertionSpecValidator.class);
     /**
      * Instantiates a new saml2 allow no subject assertion spec validator.
      */
     public Saml2AllowNoSubjectAssertionSpecValidator() {
         super();
     }
 
+    /* (non-Javadoc)
+     * @see gov.hhs.fha.nhinc.callback.cxf.Saml2ExchangeAuthFrameworkValidator#validateAssertion(org.apache.wss4j.common.saml.SamlAssertionWrapper)
+     */
+    @Override
+    protected void validateAssertion(final SamlAssertionWrapper samlAssertion) throws WSSecurityException {
+        try {
+            validateSubject(samlAssertion.getSaml2());
+        } catch (final WSSecurityException e) {
+            LOG.error("Unable to validate Assertion ", e);
+            throw e;
```

Comment:
Logging a full Exception (not just the message) and then immediately rethrowing is dangerous -- this is how log files end up with dozens of stack traces for one error condition.  If you must log here, consider only logging the exception **message** (e.getLocalizedMessage()), and waiting to output the full stack at the point where the exception is actually handled.